### PR TITLE
Update modeling doc strings FE -> IP

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Replaces references of feature extractors with image processors that had been missed in the first batch of changes 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/beit/modeling_beit.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "BeitConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "BeitImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "BeitImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/beit/modeling_beit.py:52]: This should actually be removed entirely.

File: src/transformers/models/beit/modeling_beit.py [hunk 2]
```diff
@@ -646,7 +646,7 @@ class PreTrainedModel
 
     @add_start_docstrings_to_model_forward(BEIT_INPUTS_DOCSTRING)
     @add_code_sample_docstrings(
-        processor_class=_FEAT_EXTRACTOR_FOR_DOC,
+        processor_class=_IMAGE_PROCESSOR_FOR_DOC,
```
Comment (sgugger) [inline src/transformers/models/beit/modeling_beit.py:649]: You don't need to pass those args anymore.

File: src/transformers/models/beit/modeling_beit.py [hunk 3]
```diff
@@ -844,7 +844,7 @@ def __init__(self, config: BeitConfig) -> None:
...
```
Comment (sgugger) [inline src/transformers/models/beit/modeling_beit.py:847]: Same here.

File: src/transformers/models/bit/modeling_bit.py [hunk 4]
```diff
@@ -46,7 +46,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "BitConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "AutoFeatureExtractor"
+_IMAGE_PROCESSOR_FOR_DOC = "AutoImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/bit/modeling_bit.py:49]: Same here.

File: src/transformers/models/convnext/modeling_convnext.py [hunk 5]
```diff
@@ -44,7 +44,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "ConvNextConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "ConvNextImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "ConvNextImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/convnext/modeling_convnext.py:47]: Same here.

File: src/transformers/models/cvt/modeling_cvt.py [hunk 6]
```diff
@@ -35,7 +35,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "CvtConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "AutoImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "AutoImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/cvt/modeling_cvt.py:38]: Same here.

File: src/transformers/models/data2vec/modeling_data2vec_vision.py [hunk 7]
```diff
@@ -48,7 +48,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "Data2VecVisionConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "BeitImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "BeitImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/data2vec/modeling_data2vec_vision.py:51]: Same here.

File: src/transformers/models/data2vec/modeling_tf_data2vec_vision.py [hunk 8]
```diff
@@ -53,7 +53,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "Data2VecVisionConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "BeitImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "BeitImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/data2vec/modeling_tf_data2vec_vision.py:56]: Same here.

File: src/transformers/models/deit/modeling_deit.py [hunk 9]
```diff
@@ -44,7 +44,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "DeiTConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "DeiTImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "DeiTImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/deit/modeling_deit.py:47]: Same here.

File: src/transformers/models/deit/modeling_tf_deit.py [hunk 10]
```diff
@@ -52,7 +52,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "DeiTConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "DeiTImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "DeiTImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/deit/modeling_tf_deit.py:55]: and there

File: src/transformers/models/dinat/modeling_dinat.py [hunk 11]
```diff
@@ -57,7 +57,7 @@ def natten2dav(*args, **kwargs):
 
 # General docstring
 _CONFIG_FOR_DOC = "DinatConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "AutoImageProcessor"
+_IMAGE_PROCESSOR_FOR_DOC = "AutoImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/dinat/modeling_dinat.py:60]: also here

File: src/transformers/models/donut/modeling_donut_swin.py [hunk 12]
```diff
@@ -43,7 +43,7 @@
 
 # General docstring
 _CONFIG_FOR_DOC = "DonutSwinConfig"
-_FEAT_EXTRACTOR_FOR_DOC = "AutoFeatureExtractor"
+_IMAGE_PROCESSOR_FOR_DOC = "AutoImageProcessor"
```
Comment (sgugger) [inline src/transformers/models/donut/modeling_donut_swin.py:46]: and here.

**Review by sgugger — COMMENTED** [review]
Note that `add_code_sample_docstrings` now ignores the `processor_class` argument, so you can just remove it entirely in all those model files. Left pointers on the first ones and then got tired of saying "Same here" ;-)

**Review by sgugger — APPROVED** [review]
Thanks for all the work on this!

**Review by NielsRogge — APPROVED** [review]
Thanks a lot!

**Pull request merged**